README.ConfigureResourceReferences.md

```diff
@@ -0,0 +1,185 @@
+    or field description mentioning `refer`, `GCP`/`google`/`cloud` resources,
+    `url`, `uri`, etc. Once you see these keywords, it's very likely that this
+    field references to gcp resource. Double-check the GCP tutorial and/or TF
+    documentation to confirm.
+
+Figure out the referenced GCP resource through reading the field description. If
+you are still unsure, contact the API team for more information. Once it's
+determined, note down the field name and the referenced GCP resource name.
+
+## Identify whether the referenced resource is TF-based or DCL-based
+
+### Turn GCP resource name or TF type name into KCC kind name
+
+Before moving on, you need to turn the referenced TF type/GCP resource names
+you've noted down into KCC kind names. Ask the KCC team for help if you still
+can't figure them out with the following instructions.
+
+1.  If you know the TF type name, you can do project-scoped search in your local
+    repository with the TF type name. If you find a hit in a service mapping
+    yaml file (under either the `config/servicemappings/` folder or the
+    `scripts/resource-autogen/generated/servicemappings/`), the corresponding
+    `kind` field in the same resource config should be the KCC kind.
+
+1.  If you know the GCP resource name, you can either use the search widget to
+    find the corresponding TF type in
+    [Terraform doc](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs)
+    with the GCP resource name, then figure out the KCC kind name following the
+    step above; or you can use the search widget to find the corresponding KCC
+    kind directly in
+    [KCC reference doc](https://cloud.google.com/config-connector/docs/reference/overview).
+
+1.  If you can't find a KCC kind with the steps above, it's possible that the
+    referenced resource is not supported in KCC. Reach out to the KCC team is
+    you are unsure. If you've determined that the KCC kind is not supported in
+    KCC, skip the next couple of sections and go to
+    [Configure reference resource in the service mappings](#configure-reference-in-the-service-mappings).
+
+### Determine whether the referenced resource is TF-based or DCL-based
+
+Based on the KCC kind names of the references you figured above, follow the
+steps to identify the referenced resource is DCL or TF based.
+
+1.  Search for the KCC kind name of the referenced resource on the
+    [crds folder](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/crds).
+
+1.  Open the CRD file containing the KCC kind name, search for `dcl2crd` and
+    `tf2crd` labels (only one should exist for a particular CRD).
+
+    1.  If `cnrm.cloud.google.com/dcl2crd: "true"` is found, then it is a
+        DCL-based resource.
+
+    1.  If `cnrm.cloud.google.com/tf2crd: "true"` is found, then it is a
+        TF-based resource.
+
+## Identify whether the reference field is a string field or a field of other types (list, etc)
```

Comment:
could we shorten this title to something like "Identify the reference field type" and then expand the description to something like "Next, you identify the type of the reference field. In most cases, the reference field is a string field, but could be another type like a list. Check the Terraform schema of the target resource..."